src/server/tcp.rs

```diff
@@ -102,11 +103,25 @@ async fn handle_new_client(
     state: ServerStateRef,
 ) -> Result<(), anyhow::Error> {
     let (version, authenticate, crypt_state) = Client::init(&mut tls_stream, server_version).await.context("init client")?;
+    let version_release = version.get_release();
 
     let (read, write) = io::split(tls_stream);
     let (tx, rx) = mpsc::channel(MAX_BANDWIDTH_IN_BYTES);
 
     let username = authenticate.get_username().to_string();
+    
+    let username_regex = Regex::new(r"^\[\d+\].*$").unwrap();
+    if !username_regex.is_match(&username) || version_release != "CitizenFX Client" {
```

Comment:
This check should be before we create our message channels.

Reply:
I believe I have done this now in commit a92e04b